### DISCUSSION TITLE
Working great, when it works, but randomly tabs will become unlocked and close themselves.

### DISCUSSION THREAD

**Discussion by TheIndieArmy** [tabwrangler/tabwrangler#465] [General]
I've had an ongoing issue where tabs that I have locked will no longer be open when returning to the PC after some time. I don't restart my PC, I don't put it to sleep or hibernate, it's always on and screen always active. However, I have yet to notice this happening when I'm actively using the PC. It's only been after I've stepped away for some time.

At this point, I feel like I'm wrangling the wrangler more than it is wrangling the tabs. I often have to go into the Tab Corral and select a bunch of tabs I want to reopen, then I have to go into the Tab Lock and select them all again (because for some reason only the Tab Corral has "select all" at the top) to lock them all. Only to have to repeat this process every few hours with a random set of different tabs the next time. 

Does anyone know how I can keep tabs locked for good? Thanks!

**Comment by TheIndieArmy** [comment]
Also, not sure if this is related, but whenever I do restart my PC or update Chrome my tab lock shows all my previously locked tabs. The only ones actually locked are any tabs in a Chrome tab group. All others and unchecked and the countdown has started on them. This is another reason why having a "Select All > Lock/Unlock" options would be nice on the tab lock.

**Comment by ssorallen** [comment]
Thank you for the report. This is likely changes in v7 to accommodate Chrome’s new APIs. I will try to reproduce this and report back in this thread. It will take me a week or two before I can investigate.

**Comment by TheIndieArmy** [comment]
Thanks for the acknowledgment. Is there a way for me to pull logs and assist in any way? I didn't see any such option in the app, but wants to ask in case it's possible.

**Comment by ssorallen** [comment]
Likely related to https://github.com/tabwrangler/tabwrangler/issues/466.

**Comment by ssorallen** [comment]
@TheIndieArmy  Could you see if Tab Wrangler is throwing any exceptions?
1. Window > Extensions
2. Toggle "Developer Mode" in upper right corner
3. Click the "service worker" link that says Inspect views service worker next to Tab Wrangler
4. Paste any red exceptions from the Console into this issue

Thank you.

**Reply by TheIndieArmy** [reply]
Thanks. I already had developer mode on and this is what it shows now, but this is current. Not necessarily after I noticed the issue occurring. So not sure if these errors are related. I will check and provide errors again when I notice the issue next as well. 

Unchecked runtime.lastError: Cannot create item with duplicate id corralTab
Unchecked runtime.lastError: Cannot create item with duplicate id separator
Unchecked runtime.lastError: Cannot create item with duplicate id lockTab
Unchecked runtime.lastError: Cannot create item with duplicate id lockDomain
2background.entry.js:1739 [checkToClose] Error: Maximum execution time is exceeded local.tabTimes
    at background.entry.js:162:19
checkToClose @ background.entry.js:1739

**Comment by ssorallen** [comment]
Thank you, that helps a lot. The following line means it was unable to acquire a lock for editing `"local.tabTimes"`, which explains why it is not closing tabs.

> 2background.entry.js:1739 [checkToClose] Error: Maximum execution time is exceeded local.tabTimes
at background.entry.js:162:19

It doesn't explain why the lock is not being released, but this helps immensely in tracking that down. I will post further updates after I take a look at the code.

**Reply by ssorallen** [reply]
I mistyped there - it doesn't mean it couldn't acquire the lock, it means the execution took too long. The work to close tabs likely isn't completing.

**Comment by TheIndieArmy** [comment]
Hmm, maybe my original post wasn't clear or I'm not understanding your previous message. My issue is that when I return after some time, some tabs that I did have locked will be unlocked and closed. I don't have any issues with closing tabs myself.

**Reply by ssorallen** [reply]
Oh right thank you. I am also investigating tabs not closing properly for some users, and that error may be the reason why.

**Comment by ssorallen** [comment]
@TheIndieArmy I am going to close this in favor of the issue that was opened, https://github.com/tabwrangler/tabwrangler/issues/462. Please post there if you are able to find any further debug information.
